tests/zquantum/core/measurement_test.py

```diff
@@ -857,3 +858,81 @@ def test_get_measurements_representing_distribution_gives_exactly_right_counts(
         counts = measurements.get_counts()
         for bitstring, probability in bitstring_distribution.distribution_dict.items():
             assert probability * number_of_samples == counts[bitstring]
+
+    @pytest.mark.parametrize(
+        "bitstring_distribution",
+        [
+            BitstringDistribution(
+                {
+                    "0011": 0.0049,
+                    "1100": 0.0049,
+                    "1111": 0.008,
+                    "0000": 0.008,
+                    "0001": 0.008,
+                    "0010": 0.008,
+                    "0100": 0.008,
+                    "1000": 0.008,
+                    "1001": 0.008,
+                    "1010": 0.008,
+                    "1110": 0.008,
+                    "1101": 0.008,
+                    "0110": 0.9102,
+                }
+            )
+        ],
+    )
+    def test_get_measurements_representing_distribution_does_not_remove_bitstrings_not_sampled(
+        self, bitstring_distribution
+    ):
+        number_of_samples = 100
+        max_number_of_trials = 100
+        cnt = 0
+        for _ in range(max_number_of_trials):
+            measurements = Measurements.get_measurements_representing_distribution(
+                bitstring_distribution, number_of_samples
+            )
```

Comment:
At the moment this test only verifies that no exception is raised. Is there any other behavior to assert upon?

Reply:
No, that is all that is checked here. The reason for adding this test is that before the bugfix, it would raise an Exception for trying to remove a non-existing element from a list. In case further modification of the code reintroduce this bug, it would be caught here.

Reply:
would it make sense to rename the test case to reflect that, e.g. `test_get_measurements_representing_distribution_doesnt_raise`?  This would remove the need for the comment :)

Reply:
Good point! Done in 005181dd6e9aa6c054525a3aff8dd0168bf48235